djhtml/__main__.py

```diff
@@ -1,210 +1,152 @@
-        sys.exit("Will not modify files in-place without -i option")
-
-    for input_filename in _generate_files(args.input_filenames, target_extension):
+    for filename in _generate_filenames(options.input_filenames, suffixes):
         # Read input file
         try:
-            input_file = (
-                sys.stdin if input_filename == "-" else open(input_filename, "r")
-            )
+            input_file = sys.stdin if filename == "-" else open(filename, "r")
             source = input_file.read()
+            input_file.close()
         except Exception as e:
             problematic_files += 1
-            if not args.quiet:
-                print(f"Error opening {input_filename}: {e}", file=sys.stderr)
+            _error(e)
             continue
 
         # Indent input file
         try:
-            if args.debug:
+            if options.debug:
                 print(Mode(source).debug())
                 sys.exit()
-            result = Mode(source).indent(args.tabwidth)
+            result = Mode(source).indent(options.tabwidth)
         except SyntaxError as e:
             problematic_files += 1
-            if not args.quiet:
-                print(
-                    f"Syntax error in {input_file.name}:"
-                    f" {str(e) or e.__class__.__name__}",
-                    file=sys.stderr,
-                )
+            _error(f"Syntax error in {filename}: {str(e) or e.__class__.__name__}")
             continue
         except Exception:
-            print(
-                f"\nFatal error while processing {input_file.name}\n\n"
+            _error(
+                f"Fatal error while processing {filename}\n\n"
                 "    If you have time and are using the latest version, we\n"
                 "    would very much appreciate if you opened an issue on\n"
-                "    https://github.com/rtts/djhtml/issues\n",
-                file=sys.stderr,
+                "    https://github.com/rtts/djhtml/issues\n"
             )
             raise
-        finally:
-            input_file.close()
 
-        changed = verify_changed(source, result)
-
-        # Print to stdout and exit
-        if not args.in_place and not args.check and args.output_file == "-":
-            if not args.quiet:
-                print(result, end="")
-            sys.exit(1 if args.check and changed else 0)
+        changed = _verify_changed(source, result)
+        if changed:
+            changed_files += 1
+        else:
+            unchanged_files += 1
 
         # Write output file
-        if changed and args.check:
-            changed_files += 1
+        if filename == "-":
+            if not options.check:
+                print(result, end="")
         elif changed:
-            output_filename = input_file.name if args.in_place else args.output_file
             try:
-                output_file = open(output_filename, "w")
+                output_file = open(filename, "w")
```

Comment:
Wouldn't the more Pythonic syntax be:

```
with open(filename, "w") as output_file:
    output_file.write(result)
```

Reply:
Yes, it would. I was holding it off because when opening I also didn't use this syntax, and I wanted to be consistent.

While fixing this, however, I encountered a bug much more severe: even with the `--check` argument the output files were written! :scream: 

Both have now been fixed.